### PR TITLE
Prepare for plugin dynamic

### PR DESCRIPTION
**Pull request by VladRassokhin**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -62,6 +62,11 @@ tasks {
     }
   }
 
+  runIde {
+    jvmArgs("-Dide.plugins.snapshot.on.unload.fail=true", "-XX:+UnlockDiagnosticVMOptions")
+    autoReloadPlugins.set(true)
```
Comment (ForNeVeR) [inline build.gradle.kts:67]: If I understand correctly, currently this doesn't work on Linux and macOS, right? So, let's maybe make this conditional on the current OS, or disable it until #121 is resolved? What do you think?

File: src/main/kotlin/com/intellij/plugin/powershell/lang/lsp/ide/EditorEventManager.kt [hunk 2]
```diff
@@ -58,10 +58,9 @@ class EditorEventManager(private val project: Project, private val editor: Edito
   fun getDiagnostics(): List<Diagnostic> = diagnosticsInfo
 
   companion object {
-    private val uriToManager = mutableMapOf<URI, EditorEventManager>()
-    private val editorToManager = mutableMapOf<Editor, EditorEventManager>()
+    private val EDITOR_EVENT_MANAGER_KEY: Key<EditorEventManager> = Key("Powershell.EditorEventManager")
     fun forEditor(editor: Editor): EditorEventManager? {
-      return editorToManager[editor]
+      return editor.getUserData(EDITOR_EVENT_MANAGER_KEY)
```
Comment (ForNeVeR) [inline src/main/kotlin/com/intellij/plugin/powershell/lang/lsp/ide/EditorEventManager.kt:63]: Hey, will this work correctly if the plugin is unloaded? Is there some special treatment in the platform for the user data keys introduced by the plugins?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/intellij/plugin/powershell/lang/lsp/languagehost/EditorServicesLanguageHostStarter.kt [hunk 3]
```diff
@@ -164,7 +164,7 @@ open class EditorServicesLanguageHostStarter(protected val myProject: Project) :
   }
 
   private fun getSessionCount(): Int {
-    return ++sessionCount - 1
+    return sessionCount.getAndIncrement()
```
Comment (ForNeVeR) [inline src/main/kotlin/com/intellij/plugin/powershell/lang/lsp/languagehost/EditorServicesLanguageHostStarter.kt:167]: Ha, a good catch! 👍

File: build.gradle.kts (hunk 1 above)
Comment (VladRassokhin) [reply build.gradle.kts:67]: I've tested on macOS, as I've no other OS. :)
While the plugin is marked as `requires-restart=true`, those options would be no-op, AFAIU.

File: src/main/kotlin/com/intellij/plugin/powershell/lang/lsp/languagehost/EditorServicesLanguageHostStarter.kt (hunk 3 above)
Comment (VladRassokhin) [reply src/main/kotlin/com/intellij/plugin/powershell/lang/lsp/languagehost/EditorServicesLanguageHostStarter.kt:167]: I've just grepped for 'volatile' ;)

File: src/main/kotlin/com/intellij/plugin/powershell/lang/lsp/ide/EditorEventManager.kt (hunk 2 above)
Comment (VladRassokhin) [reply src/main/kotlin/com/intellij/plugin/powershell/lang/lsp/ide/EditorEventManager.kt:63]: No idea :)
But if we remove user data in `dispose`, it should be safe.

File: build.gradle.kts (hunk 1 above)
Comment (ForNeVeR) [reply build.gradle.kts:67]: Ok then, I'll check.

File: src/main/kotlin/com/intellij/plugin/powershell/lang/lsp/ide/EditorEventManager.kt (hunk 2 above)
Comment (ForNeVeR) [reply src/main/kotlin/com/intellij/plugin/powershell/lang/lsp/ide/EditorEventManager.kt:63]: Ok, I'll take another look at the related code.

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**